src/AzureAppConfigurationImpl.ts

```diff
@@ -55,23 +83,80 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
             };
             if (this.requestTracingEnabled) {
                 listOptions.requestOptions = {
-                    customHeaders: this.customHeaders()
+                    customHeaders: this.customHeaders(requestType)
                 }
             }
 
             const settings = this.client.listConfigurationSettings(listOptions);
 
             for await (const setting of settings) {
                 if (setting.key) {
-                    const [key, value] = await this.processAdapters(setting);
-                    const trimmedKey = this.keyWithPrefixesTrimmed(key);
-                    keyValues.push([trimmedKey, value]);
+                    const keyValuePair = await this.processKeyValues(setting);
+                    keyValues.push(keyValuePair);
+                }
+                // update etag of sentinels
+                const matchedSentinel = this.sentinels?.find(s => s.key === setting.key && (s.label ?? null) === setting.label); // Workaround: as undefined label represents the same with null.
+                if (matchedSentinel) {
+                    matchedSentinel.etag = setting.etag;
                 }
             }
         }
         for (const [k, v] of keyValues) {
             this.set(k, v);
         }
+        this.lastUpdateTimestamp = Date.now();
+    }
+
+    public async refresh(): Promise<void> {
+        // if no refreshOptions set, return
+        if (this.sentinels === undefined || this.sentinels.length === 0 || this.refreshIntervalInMs === undefined) {

```

Comment:
no default value for `refreshIntervalInMs`?

Reply:
default value is 30s. Currently it's assigned in the constructor if refreshOptions is specified. Otherwise it's undefined.

Reply:
And to confirm, we are supporting collection monitoring for now, at least not included in this PR. Is this true?

Reply:
In this PR, the supported refresh is sentinel-based. You pass `watchedSettings` as a list of sentinels, the provider periodically calls getConfigurationSetting api to check if any of them is modified. If modification is detected, the provider then re-load all the configuration settings to update the Map.